HPX master breaks Phylanx master

CircleCI shows the current master as passing but the new HPX master breaks it.

Build output:
```
# cmake --build /build --target tests -- -j1
[ 13%] Built target phylanx_component
[ 14%] Built target fileio_primitive
[ 14%] Built target primitives.fileio_dir.fileio_plugin
[ 14%] Built target primitives.fileio_dir
[ 15%] Built target solvers_primitive
[ 15%] Built target primitives.solvers_dir.solvers_plugin
[ 15%] Built target primitives.solvers_dir
[ 16%] Built target arithmetics_primitive
[ 16%] Built target primitives.arithmetics_dir.arithmetics_plugin
[ 16%] Built target primitives.arithmetics_dir
[ 23%] Built target matrixops_primitive
[ 23%] Built target primitives.matrixops_dir.matrixops_plugin
[ 23%] Built target primitives.matrixops_dir
[ 24%] Built target algorithms_primitive
[ 24%] Built target primitives.algorithms_dir.algorithms_plugin
[ 24%] Built target primitives.algorithms_dir
[ 28%] Built target controls_primitive
[ 28%] Built target primitives.controls_dir.controls_plugin
[ 28%] Built target primitives.controls_dir
[ 29%] Built target listops_primitive
[ 29%] Built target primitives.listops_dir.listops_plugin
[ 29%] Built target primitives.listops_dir
[ 34%] Built target booleans_primitive
[ 34%] Built target primitives.booleans_dir.booleans_plugin
[ 34%] Built target primitives.booleans_dir
[ 34%] Built target primitives
[ 34%] Building CXX object tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/remote_run.cpp.o
In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
In file included from /phylanx/phylanx/ast/node.hpp:12:
In file included from /phylanx/phylanx/ir/node_data.hpp:13:
In file included from /usr/local/include/hpx/include/serialization.hpp:23:
In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:15:
/usr/local/include/hpx/runtime/actions/plain_action.hpp:80:22: error: too many arguments to function call, expected 4, have 5
            return F(std::forward<Ts>(vs)...);
                   ~ ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/hpx/runtime/actions/basic_action.hpp:225:33: note: in instantiation of function template specialization 'hpx::actions::basic_action_impl<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), &phylanx::execution_tree::primitives::locality_id,
      locality_id_action>::invoke<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
                return Derived::invoke(lva, comptype, std::forward<Ts>(vs)...);
                                ^
/usr/local/include/hpx/runtime/actions/basic_action.hpp:205:24: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::invoke<std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const
      std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
                return invoke(
                       ^
/usr/local/include/hpx/runtime/actions/basic_action.hpp:351:20: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::operator()<std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const
      std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
            return invoker()(lva, comptype, std::forward<Ts>(vs)...);
                   ^
/usr/local/include/hpx/lcos/detail/async_implementations.hpp:150:33: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::execute_function<std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const
      std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
                        Action::execute_function(
                                ^
/usr/local/include/hpx/lcos/detail/async_implementations.hpp:338:24: note: in instantiation of function template specialization 'hpx::detail::sync_local_invoke<locality_id_action, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>
      >::call<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
                    >::call(id, std::move(addr), std::forward<Ts>(vs)...);
                       ^
/usr/local/include/hpx/lcos/detail/async_implementations.hpp:377:21: note: (skipping 13 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                f = async_local_impl<Action>(policy, id, addr, r,
                    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
        : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
          ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
      >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
      return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
             ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
      const std::__cxx11::basic_string<char> &>' requested here
      return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
                  ^
/phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
            std::make_shared<Primitive>(std::move(args), name, codename));
                 ^
/phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
            &create_primitive<generic_function<locality_id_action>>,
             ^
In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
In file included from /phylanx/phylanx/ast/node.hpp:12:
In file included from /phylanx/phylanx/ir/node_data.hpp:13:
In file included from /usr/local/include/hpx/include/serialization.hpp:23:
In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:15:
/usr/local/include/hpx/runtime/actions/plain_action.hpp:80:22: error: too many arguments to function call, expected 4, have 5
            return F(std::forward<Ts>(vs)...);
                   ~ ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/hpx/runtime/actions/basic_action.hpp:225:33: note: in instantiation of function template specialization 'hpx::actions::basic_action_impl<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), &phylanx::execution_tree::primitives::locality_id,
      locality_id_action>::invoke<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
                return Derived::invoke(lva, comptype, std::forward<Ts>(vs)...);
                                ^
/usr/local/include/hpx/runtime/actions/basic_action.hpp:205:24: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::invoke<std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
      std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
                return invoke(
                       ^
/usr/local/include/hpx/lcos/detail/async_implementations.hpp:318:49: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::operator()<std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
      std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
            return get_remote_result_type::call(typename Action::invoker()(
                                                ^
/usr/local/include/hpx/util/invoke.hpp:36:21: note: in instantiation of function template specialization 'hpx::detail::action_invoker<locality_id_action>::operator()<std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
      std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
                    std::forward<F>(f)(std::forward<Ts>(vs)...);
                    ^
/usr/local/include/hpx/async_launch_policy_dispatch.hpp:61:44: note: in instantiation of member function 'hpx::util::detail::deferred<hpx::detail::action_invoker<locality_id_action>, unsigned long, int, std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
      std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>::operator()' requested here
            return lcos::make_ready_future(f());
                                           ^
/usr/local/include/hpx/async_launch_policy_dispatch.hpp:108:32: note: (skipping 21 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                return detail::call_sync(
                               ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
        : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
          ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
      >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
      return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
             ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
      const std::__cxx11::basic_string<char> &>' requested here
      return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
                  ^
/phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
            std::make_shared<Primitive>(std::move(args), name, codename));
                 ^
/phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
            &create_primitive<generic_function<locality_id_action>>,
             ^
In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
In file included from /phylanx/phylanx/ast/node.hpp:12:
In file included from /phylanx/phylanx/ir/node_data.hpp:13:
In file included from /usr/local/include/hpx/include/serialization.hpp:23:
In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:12:
In file included from /usr/local/include/hpx/lcos/async.hpp:11:
In file included from /usr/local/include/hpx/lcos/detail/async_implementations.hpp:12:
In file included from /usr/local/include/hpx/lcos/packaged_action.hpp:11:
In file included from /usr/local/include/hpx/lcos/promise.hpp:12:
In file included from /usr/local/include/hpx/lcos/detail/promise_base.hpp:13:
In file included from /usr/local/include/hpx/lcos/detail/promise_lco.hpp:12:
In file included from /usr/local/include/hpx/lcos/base_lco_with_value.hpp:12:
In file included from /usr/local/include/hpx/lcos/base_lco.hpp:10:
In file included from /usr/local/include/hpx/runtime/actions/basic_action.hpp:21:
In file included from /usr/local/include/hpx/runtime/actions/transfer_action.hpp:14:
/usr/local/include/hpx/runtime/actions/transfer_base_action.hpp:194:13: error: no matching constructor for initialization of 'hpx::actions::transfer_base_action<locality_id_action>::arguments_type' (aka
      'hpx::util::tuple<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >')
          , arguments_(std::forward<Ts>(vs)...)
            ^          ~~~~~~~~~~~~~~~~~~~~
/usr/local/include/hpx/runtime/actions/transfer_continuation_action.hpp:118:9: note: in instantiation of function template specialization
      'hpx::actions::transfer_base_action<locality_id_action>::transfer_base_action<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>'
      requested here
      : base_type(std::forward<Ts>(vs)...)
        ^
/usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:52:29: note: in instantiation of function template specialization 'hpx::actions::transfer_continuation_action<locality_id_action>::transfer_continuation_action<hpx::threads::thread_priority &,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
                        new actions::transfer_continuation_action<Action>(
                            ^
/usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:100:43: note: in instantiation of function template specialization 'hpx::parcelset::detail::create_parcel::call<locality_id_action,
      hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, hpx::threads::thread_priority &,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
                pp(detail::create_parcel::call(
                                          ^
/usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:193:17: note: in instantiation of function template specialization 'hpx::parcelset::detail::put_parcel_impl<hpx::parcelset::detail::put_parcel_handler_cb<hpx::util::function<void
      (const boost::system::error_code &, const hpx::parcelset::parcel &), false> >, hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >,
      locality_id_action &, hpx::threads::thread_priority &, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
        detail::put_parcel_impl(
                ^
/usr/local/include/hpx/runtime/applier/apply.hpp:155:24: note: in instantiation of function template specialization 'hpx::parcelset::put_parcel_cb<hpx::util::function<void (const boost::system::error_code &, const hpx::parcelset::parcel &), false>,
      hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, locality_id_action &, hpx::threads::thread_priority &,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
            parcelset::put_parcel_cb(std::move(cb), id,
                       ^
/usr/local/include/hpx/runtime/applier/apply_callback.hpp:131:28: note: (skipping 17 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
            return detail::put_parcel_cont_cb<Action>(id, std::move(addr),
                           ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
        : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
          ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
      >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
      return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
             ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
      const std::__cxx11::basic_string<char> &>' requested here
      return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
                  ^
/phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
            std::make_shared<Primitive>(std::move(args), name, codename));
                 ^
/phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
            &create_primitive<generic_function<locality_id_action>>,
             ^
/usr/local/include/hpx/util/tuple.hpp:447:48: note: candidate template ignored: requirement 'std::conditional<detail::pack<const vector<primitive_argument_type, allocator<primitive_argument_type> > &, const basic_string<char> &, const basic_string<char> &,
      eval_context>::size == 0, typename std::enable_if<!std::is_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, typename
      std::decay<vector<primitive_argument_type, allocator<primitive_argument_type> > &>::type>::value && !detail::are_tuples_compatible_not_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type,
      allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, vector<primitive_argument_type, allocator<primitive_argument_type> > &>::value, detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type>
      >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type, allocator<primitive_argument_type> > &> &&> >::type,
      detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type,
      allocator<primitive_argument_type> > &, const vector<primitive_argument_type, allocator<primitive_argument_type> > &, const basic_string<char> &, const basic_string<char> &, eval_context> &&> >::type::value' was not satisfied [with U =
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, Us = <const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>]
        explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(U&& v, Us&&... vs)
                                               ^
/usr/local/include/hpx/util/tuple.hpp:416:39: note: candidate constructor template not viable: requires 0 arguments, but 5 were provided
        HPX_CONSTEXPR HPX_HOST_DEVICE tuple()
                                      ^
/usr/local/include/hpx/util/tuple.hpp:488:39: note: candidate constructor template not viable: requires single argument 'other', but 5 arguments were provided
        HPX_CONSTEXPR HPX_HOST_DEVICE tuple(UTuple&& other)
                                      ^
/usr/local/include/hpx/util/tuple.hpp:423:48: note: candidate constructor not viable: requires 4 arguments, but 5 were provided
        explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(Ts const&... vs)
                                               ^
/usr/local/include/hpx/util/tuple.hpp:455:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
        tuple(tuple const&) = default;
        ^
/usr/local/include/hpx/util/tuple.hpp:460:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
        tuple(tuple&&) = default;
        ^
In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
In file included from /phylanx/phylanx/ast/node.hpp:12:
In file included from /phylanx/phylanx/ir/node_data.hpp:13:
In file included from /usr/local/include/hpx/include/serialization.hpp:23:
In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:12:
In file included from /usr/local/include/hpx/lcos/async.hpp:11:
In file included from /usr/local/include/hpx/lcos/detail/async_implementations.hpp:12:
In file included from /usr/local/include/hpx/lcos/packaged_action.hpp:11:
In file included from /usr/local/include/hpx/lcos/promise.hpp:12:
In file included from /usr/local/include/hpx/lcos/detail/promise_base.hpp:13:
In file included from /usr/local/include/hpx/lcos/detail/promise_lco.hpp:12:
In file included from /usr/local/include/hpx/lcos/base_lco_with_value.hpp:12:
In file included from /usr/local/include/hpx/lcos/base_lco.hpp:10:
In file included from /usr/local/include/hpx/runtime/actions/basic_action.hpp:21:
In file included from /usr/local/include/hpx/runtime/actions/transfer_action.hpp:14:
/usr/local/include/hpx/runtime/actions/transfer_base_action.hpp:194:13: error: no matching constructor for initialization of 'hpx::actions::transfer_base_action<locality_id_action>::arguments_type' (aka
      'hpx::util::tuple<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >')
          , arguments_(std::forward<Ts>(vs)...)
            ^          ~~~~~~~~~~~~~~~~~~~~
/usr/local/include/hpx/runtime/actions/transfer_continuation_action.hpp:118:9: note: in instantiation of function template specialization
      'hpx::actions::transfer_base_action<locality_id_action>::transfer_base_action<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
      : base_type(std::forward<Ts>(vs)...)
        ^
/usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:52:29: note: in instantiation of function template specialization 'hpx::actions::transfer_continuation_action<locality_id_action>::transfer_continuation_action<hpx::threads::thread_priority,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >,
      std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
                        new actions::transfer_continuation_action<Action>(
                            ^
/usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:142:64: note: in instantiation of function template specialization 'hpx::parcelset::detail::create_parcel::call<locality_id_action,
      hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, hpx::threads::thread_priority,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >,
      std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
                                    pp_(detail::create_parcel::call(
                                                               ^
/usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:193:17: note: in instantiation of function template specialization 'hpx::parcelset::detail::put_parcel_impl<hpx::parcelset::detail::put_parcel_handler_cb<hpx::util::function<void
      (const boost::system::error_code &, const hpx::parcelset::parcel &), false> >, hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >,
      locality_id_action &, hpx::threads::thread_priority &, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
        detail::put_parcel_impl(
                ^
/usr/local/include/hpx/runtime/applier/apply.hpp:155:24: note: in instantiation of function template specialization 'hpx::parcelset::put_parcel_cb<hpx::util::function<void (const boost::system::error_code &, const hpx::parcelset::parcel &), false>,
      hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, locality_id_action &, hpx::threads::thread_priority &,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
      std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
            parcelset::put_parcel_cb(std::move(cb), id,
                       ^
/usr/local/include/hpx/runtime/applier/apply_callback.hpp:131:28: note: (skipping 17 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
            return detail::put_parcel_cont_cb<Action>(id, std::move(addr),
                           ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
        : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
          ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
      >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
      std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
      return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
             ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
      const std::__cxx11::basic_string<char> &>' requested here
      return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
                  ^
/phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
            std::make_shared<Primitive>(std::move(args), name, codename));
                 ^
/phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
            &create_primitive<generic_function<locality_id_action>>,
             ^
/usr/local/include/hpx/util/tuple.hpp:447:48: note: candidate template ignored: requirement 'std::conditional<detail::pack<vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char>, eval_context>::size == 0, typename
      std::enable_if<!std::is_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, typename
      std::decay<vector<primitive_argument_type, allocator<primitive_argument_type> > >::type>::value && !detail::are_tuples_compatible_not_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type,
      allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, vector<primitive_argument_type, allocator<primitive_argument_type> > &&>::value, detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type>
      >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type, allocator<primitive_argument_type> > > &&> >::type,
      detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type,
      allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char>, eval_context> &&> >::type::value' was not satisfied [with U =
      std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, Us = <std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type>
      >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>]
        explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(U&& v, Us&&... vs)
                                               ^
/usr/local/include/hpx/util/tuple.hpp:416:39: note: candidate constructor template not viable: requires 0 arguments, but 5 were provided
        HPX_CONSTEXPR HPX_HOST_DEVICE tuple()
                                      ^
/usr/local/include/hpx/util/tuple.hpp:488:39: note: candidate constructor template not viable: requires single argument 'other', but 5 arguments were provided
        HPX_CONSTEXPR HPX_HOST_DEVICE tuple(UTuple&& other)
                                      ^
/usr/local/include/hpx/util/tuple.hpp:423:48: note: candidate constructor not viable: requires 4 arguments, but 5 were provided
        explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(Ts const&... vs)
                                               ^
/usr/local/include/hpx/util/tuple.hpp:455:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
        tuple(tuple const&) = default;
        ^
/usr/local/include/hpx/util/tuple.hpp:460:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
        tuple(tuple&&) = default;
        ^
4 errors generated.
tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/build.make:62: recipe for target 'tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/remote_run.cpp.o' failed
make[3]: *** [tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/remote_run.cpp.o] Error 1
CMakeFiles/Makefile2:9632: recipe for target 'tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/all' failed
make[2]: *** [tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/all] Error 2
CMakeFiles/Makefile2:300: recipe for target 'CMakeFiles/tests.dir/rule' failed
make[1]: *** [CMakeFiles/tests.dir/rule] Error 2
Makefile:266: recipe for target 'tests' failed
make: *** [tests] Error 2
```

Yes, the buildbot has been failing for about 11 days now…

http://ktau.nic.uoregon.edu:8020/#/builders/8/builds/289 <http://ktau.nic.uoregon.edu:8020/#/builders/8/builds/289>

Kevin

> On Nov 28, 2018, at 1:16 PM, Parsa Amini <notifications@github.com> wrote:
> 
> CircleCI shows the current master as passing but the new HPX master breaks it.
> 
> Build output:
> 
> # cmake --build /build --target tests -- -j1
> [ 13%] Built target phylanx_component
> [ 14%] Built target fileio_primitive
> [ 14%] Built target primitives.fileio_dir.fileio_plugin
> [ 14%] Built target primitives.fileio_dir
> [ 15%] Built target solvers_primitive
> [ 15%] Built target primitives.solvers_dir.solvers_plugin
> [ 15%] Built target primitives.solvers_dir
> [ 16%] Built target arithmetics_primitive
> [ 16%] Built target primitives.arithmetics_dir.arithmetics_plugin
> [ 16%] Built target primitives.arithmetics_dir
> [ 23%] Built target matrixops_primitive
> [ 23%] Built target primitives.matrixops_dir.matrixops_plugin
> [ 23%] Built target primitives.matrixops_dir
> [ 24%] Built target algorithms_primitive
> [ 24%] Built target primitives.algorithms_dir.algorithms_plugin
> [ 24%] Built target primitives.algorithms_dir
> [ 28%] Built target controls_primitive
> [ 28%] Built target primitives.controls_dir.controls_plugin
> [ 28%] Built target primitives.controls_dir
> [ 29%] Built target listops_primitive
> [ 29%] Built target primitives.listops_dir.listops_plugin
> [ 29%] Built target primitives.listops_dir
> [ 34%] Built target booleans_primitive
> [ 34%] Built target primitives.booleans_dir.booleans_plugin
> [ 34%] Built target primitives.booleans_dir
> [ 34%] Built target primitives
> [ 34%] Building CXX object tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/remote_run.cpp.o
> In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
> In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
> In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
> In file included from /phylanx/phylanx/ast/node.hpp:12:
> In file included from /phylanx/phylanx/ir/node_data.hpp:13:
> In file included from /usr/local/include/hpx/include/serialization.hpp:23:
> In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
> In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
> In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:15:
> /usr/local/include/hpx/runtime/actions/plain_action.hpp:80:22: error: too many arguments to function call, expected 4, have 5
>             return F(std::forward<Ts>(vs)...);
>                    ~ ^~~~~~~~~~~~~~~~~~~~
> /usr/local/include/hpx/runtime/actions/basic_action.hpp:225:33: note: in instantiation of function template specialization 'hpx::actions::basic_action_impl<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), &phylanx::execution_tree::primitives::locality_id,
>       locality_id_action>::invoke<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>                 return Derived::invoke(lva, comptype, std::forward<Ts>(vs)...);
>                                 ^
> /usr/local/include/hpx/runtime/actions/basic_action.hpp:205:24: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::invoke<std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const
>       std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>                 return invoke(
>                        ^
> /usr/local/include/hpx/runtime/actions/basic_action.hpp:351:20: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::operator()<std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const
>       std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>             return invoker()(lva, comptype, std::forward<Ts>(vs)...);
>                    ^
> /usr/local/include/hpx/lcos/detail/async_implementations.hpp:150:33: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::execute_function<std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const
>       std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>                         Action::execute_function(
>                                 ^
> /usr/local/include/hpx/lcos/detail/async_implementations.hpp:338:24: note: in instantiation of function template specialization 'hpx::detail::sync_local_invoke<locality_id_action, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>
>       >::call<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>                     >::call(id, std::move(addr), std::forward<Ts>(vs)...);
>                        ^
> /usr/local/include/hpx/lcos/detail/async_implementations.hpp:377:21: note: (skipping 13 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
>                 f = async_local_impl<Action>(policy, id, addr, r,
>                     ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>         : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
>           ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
>       >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>       return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
>              ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
>       const std::__cxx11::basic_string<char> &>' requested here
>       return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
>                   ^
> /phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>             std::make_shared<Primitive>(std::move(args), name, codename));
>                  ^
> /phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
>             &create_primitive<generic_function<locality_id_action>>,
>              ^
> In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
> In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
> In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
> In file included from /phylanx/phylanx/ast/node.hpp:12:
> In file included from /phylanx/phylanx/ir/node_data.hpp:13:
> In file included from /usr/local/include/hpx/include/serialization.hpp:23:
> In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
> In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
> In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:15:
> /usr/local/include/hpx/runtime/actions/plain_action.hpp:80:22: error: too many arguments to function call, expected 4, have 5
>             return F(std::forward<Ts>(vs)...);
>                    ~ ^~~~~~~~~~~~~~~~~~~~
> /usr/local/include/hpx/runtime/actions/basic_action.hpp:225:33: note: in instantiation of function template specialization 'hpx::actions::basic_action_impl<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (*)(const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), &phylanx::execution_tree::primitives::locality_id,
>       locality_id_action>::invoke<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>                 return Derived::invoke(lva, comptype, std::forward<Ts>(vs)...);
>                                 ^
> /usr/local/include/hpx/runtime/actions/basic_action.hpp:205:24: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::invoke<std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
>       std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>                 return invoke(
>                        ^
> /usr/local/include/hpx/lcos/detail/async_implementations.hpp:318:49: note: in instantiation of function template specialization 'hpx::actions::basic_action<hpx::actions::detail::plain_function, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> (const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &), locality_id_action>::invoker::operator()<std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
>       std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>             return get_remote_result_type::call(typename Action::invoker()(
>                                                 ^
> /usr/local/include/hpx/util/invoke.hpp:36:21: note: in instantiation of function template specialization 'hpx::detail::action_invoker<locality_id_action>::operator()<std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
>       std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>                     std::forward<F>(f)(std::forward<Ts>(vs)...);
>                     ^
> /usr/local/include/hpx/async_launch_policy_dispatch.hpp:61:44: note: in instantiation of member function 'hpx::util::detail::deferred<hpx::detail::action_invoker<locality_id_action>, unsigned long, int, std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>,
>       std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>::operator()' requested here
>             return lcos::make_ready_future(f());
>                                            ^
> /usr/local/include/hpx/async_launch_policy_dispatch.hpp:108:32: note: (skipping 21 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
>                 return detail::call_sync(
>                                ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>         : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
>           ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
>       >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>       return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
>              ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
>       const std::__cxx11::basic_string<char> &>' requested here
>       return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
>                   ^
> /phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>             std::make_shared<Primitive>(std::move(args), name, codename));
>                  ^
> /phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
>             &create_primitive<generic_function<locality_id_action>>,
>              ^
> In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
> In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
> In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
> In file included from /phylanx/phylanx/ast/node.hpp:12:
> In file included from /phylanx/phylanx/ir/node_data.hpp:13:
> In file included from /usr/local/include/hpx/include/serialization.hpp:23:
> In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
> In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
> In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:12:
> In file included from /usr/local/include/hpx/lcos/async.hpp:11:
> In file included from /usr/local/include/hpx/lcos/detail/async_implementations.hpp:12:
> In file included from /usr/local/include/hpx/lcos/packaged_action.hpp:11:
> In file included from /usr/local/include/hpx/lcos/promise.hpp:12:
> In file included from /usr/local/include/hpx/lcos/detail/promise_base.hpp:13:
> In file included from /usr/local/include/hpx/lcos/detail/promise_lco.hpp:12:
> In file included from /usr/local/include/hpx/lcos/base_lco_with_value.hpp:12:
> In file included from /usr/local/include/hpx/lcos/base_lco.hpp:10:
> In file included from /usr/local/include/hpx/runtime/actions/basic_action.hpp:21:
> In file included from /usr/local/include/hpx/runtime/actions/transfer_action.hpp:14:
> /usr/local/include/hpx/runtime/actions/transfer_base_action.hpp:194:13: error: no matching constructor for initialization of 'hpx::actions::transfer_base_action<locality_id_action>::arguments_type' (aka
>       'hpx::util::tuple<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >')
>           , arguments_(std::forward<Ts>(vs)...)
>             ^          ~~~~~~~~~~~~~~~~~~~~
> /usr/local/include/hpx/runtime/actions/transfer_continuation_action.hpp:118:9: note: in instantiation of function template specialization
>       'hpx::actions::transfer_base_action<locality_id_action>::transfer_base_action<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>'
>       requested here
>       : base_type(std::forward<Ts>(vs)...)
>         ^
> /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:52:29: note: in instantiation of function template specialization 'hpx::actions::transfer_continuation_action<locality_id_action>::transfer_continuation_action<hpx::threads::thread_priority &,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>                         new actions::transfer_continuation_action<Action>(
>                             ^
> /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:100:43: note: in instantiation of function template specialization 'hpx::parcelset::detail::create_parcel::call<locality_id_action,
>       hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, hpx::threads::thread_priority &,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>                 pp(detail::create_parcel::call(
>                                           ^
> /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:193:17: note: in instantiation of function template specialization 'hpx::parcelset::detail::put_parcel_impl<hpx::parcelset::detail::put_parcel_handler_cb<hpx::util::function<void
>       (const boost::system::error_code &, const hpx::parcelset::parcel &), false> >, hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >,
>       locality_id_action &, hpx::threads::thread_priority &, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>         detail::put_parcel_impl(
>                 ^
> /usr/local/include/hpx/runtime/applier/apply.hpp:155:24: note: in instantiation of function template specialization 'hpx::parcelset::put_parcel_cb<hpx::util::function<void (const boost::system::error_code &, const hpx::parcelset::parcel &), false>,
>       hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, locality_id_action &, hpx::threads::thread_priority &,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>             parcelset::put_parcel_cb(std::move(cb), id,
>                        ^
> /usr/local/include/hpx/runtime/applier/apply_callback.hpp:131:28: note: (skipping 17 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
>             return detail::put_parcel_cont_cb<Action>(id, std::move(addr),
>                            ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>         : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
>           ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
>       >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>       return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
>              ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
>       const std::__cxx11::basic_string<char> &>' requested here
>       return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
>                   ^
> /phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>             std::make_shared<Primitive>(std::move(args), name, codename));
>                  ^
> /phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
>             &create_primitive<generic_function<locality_id_action>>,
>              ^
> /usr/local/include/hpx/util/tuple.hpp:447:48: note: candidate template ignored: requirement 'std::conditional<detail::pack<const vector<primitive_argument_type, allocator<primitive_argument_type> > &, const basic_string<char> &, const basic_string<char> &,
>       eval_context>::size == 0, typename std::enable_if<!std::is_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, typename
>       std::decay<vector<primitive_argument_type, allocator<primitive_argument_type> > &>::type>::value && !detail::are_tuples_compatible_not_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type,
>       allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, vector<primitive_argument_type, allocator<primitive_argument_type> > &>::value, detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type>
>       >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type, allocator<primitive_argument_type> > &> &&> >::type,
>       detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type,
>       allocator<primitive_argument_type> > &, const vector<primitive_argument_type, allocator<primitive_argument_type> > &, const basic_string<char> &, const basic_string<char> &, eval_context> &&> >::type::value' was not satisfied [with U =
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, Us = <const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>]
>         explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(U&& v, Us&&... vs)
>                                                ^
> /usr/local/include/hpx/util/tuple.hpp:416:39: note: candidate constructor template not viable: requires 0 arguments, but 5 were provided
>         HPX_CONSTEXPR HPX_HOST_DEVICE tuple()
>                                       ^
> /usr/local/include/hpx/util/tuple.hpp:488:39: note: candidate constructor template not viable: requires single argument 'other', but 5 arguments were provided
>         HPX_CONSTEXPR HPX_HOST_DEVICE tuple(UTuple&& other)
>                                       ^
> /usr/local/include/hpx/util/tuple.hpp:423:48: note: candidate constructor not viable: requires 4 arguments, but 5 were provided
>         explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(Ts const&... vs)
>                                                ^
> /usr/local/include/hpx/util/tuple.hpp:455:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
>         tuple(tuple const&) = default;
>         ^
> /usr/local/include/hpx/util/tuple.hpp:460:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
>         tuple(tuple&&) = default;
>         ^
> In file included from /phylanx/tests/unit/distributed/remote_run.cpp:7:
> In file included from /phylanx/phylanx/execution_tree/primitives/generic_function.hpp:10:
> In file included from /phylanx/phylanx/execution_tree/primitives/base_primitive.hpp:12:
> In file included from /phylanx/phylanx/ast/node.hpp:12:
> In file included from /phylanx/phylanx/ir/node_data.hpp:13:
> In file included from /usr/local/include/hpx/include/serialization.hpp:23:
> In file included from /usr/local/include/hpx/runtime/serialization/partitioned_vector.hpp:10:
> In file included from /usr/local/include/hpx/components/containers/partitioned_vector/partitioned_vector_decl.hpp:16:
> In file included from /usr/local/include/hpx/runtime/components/copy_component.hpp:12:
> In file included from /usr/local/include/hpx/lcos/async.hpp:11:
> In file included from /usr/local/include/hpx/lcos/detail/async_implementations.hpp:12:
> In file included from /usr/local/include/hpx/lcos/packaged_action.hpp:11:
> In file included from /usr/local/include/hpx/lcos/promise.hpp:12:
> In file included from /usr/local/include/hpx/lcos/detail/promise_base.hpp:13:
> In file included from /usr/local/include/hpx/lcos/detail/promise_lco.hpp:12:
> In file included from /usr/local/include/hpx/lcos/base_lco_with_value.hpp:12:
> In file included from /usr/local/include/hpx/lcos/base_lco.hpp:10:
> In file included from /usr/local/include/hpx/runtime/actions/basic_action.hpp:21:
> In file included from /usr/local/include/hpx/runtime/actions/transfer_action.hpp:14:
> /usr/local/include/hpx/runtime/actions/transfer_base_action.hpp:194:13: error: no matching constructor for initialization of 'hpx::actions::transfer_base_action<locality_id_action>::arguments_type' (aka
>       'hpx::util::tuple<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >')
>           , arguments_(std::forward<Ts>(vs)...)
>             ^          ~~~~~~~~~~~~~~~~~~~~
> /usr/local/include/hpx/runtime/actions/transfer_continuation_action.hpp:118:9: note: in instantiation of function template specialization
>       'hpx::actions::transfer_base_action<locality_id_action>::transfer_base_action<std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>       : base_type(std::forward<Ts>(vs)...)
>         ^
> /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:52:29: note: in instantiation of function template specialization 'hpx::actions::transfer_continuation_action<locality_id_action>::transfer_continuation_action<hpx::threads::thread_priority,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >,
>       std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>                         new actions::transfer_continuation_action<Action>(
>                             ^
> /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:142:64: note: in instantiation of function template specialization 'hpx::parcelset::detail::create_parcel::call<locality_id_action,
>       hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, hpx::threads::thread_priority,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >,
>       std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>' requested here
>                                     pp_(detail::create_parcel::call(
>                                                                ^
> /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:193:17: note: in instantiation of function template specialization 'hpx::parcelset::detail::put_parcel_impl<hpx::parcelset::detail::put_parcel_handler_cb<hpx::util::function<void
>       (const boost::system::error_code &, const hpx::parcelset::parcel &), false> >, hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >,
>       locality_id_action &, hpx::threads::thread_priority &, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>         detail::put_parcel_impl(
>                 ^
> /usr/local/include/hpx/runtime/applier/apply.hpp:155:24: note: in instantiation of function template specialization 'hpx::parcelset::put_parcel_cb<hpx::util::function<void (const boost::system::error_code &, const hpx::parcelset::parcel &), false>,
>       hpx::actions::typed_continuation<hpx::lcos::future<phylanx::execution_tree::primitive_argument_type>, hpx::lcos::future<phylanx::execution_tree::primitive_argument_type> >, locality_id_action &, hpx::threads::thread_priority &,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::vector<phylanx::execution_tree::primitive_argument_type,
>       std::allocator<phylanx::execution_tree::primitive_argument_type> > &, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &, phylanx::execution_tree::eval_context>' requested here
>             parcelset::put_parcel_cb(std::move(cb), id,
>                        ^
> /usr/local/include/hpx/runtime/applier/apply_callback.hpp:131:28: note: (skipping 17 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
>             return detail::put_parcel_cont_cb<Action>(id, std::move(addr),
>                            ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:344:4: note: in instantiation of function template specialization 'std::__shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       __gnu_cxx::_S_atomic>::__shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>         : __shared_ptr<_Tp>(__tag, __a, std::forward<_Args>(__args)...)
>           ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:690:14: note: in instantiation of function template specialization 'std::shared_ptr<phylanx::execution_tree::primitives::generic_function<locality_id_action>
>       >::shared_ptr<std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const
>       std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>       return shared_ptr<_Tp>(_Sp_make_shared_tag(), __a,
>              ^
> /usr/bin/../lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/bits/shared_ptr.h:706:19: note: in instantiation of function template specialization 'std::allocate_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::allocator<phylanx::execution_tree::primitives::generic_function<locality_id_action> >, std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &,
>       const std::__cxx11::basic_string<char> &>' requested here
>       return std::allocate_shared<_Tp>(std::allocator<_Tp_nc>(),
>                   ^
> /phylanx/phylanx/execution_tree/primitives/primitive_component_base.hpp:232:18: note: in instantiation of function template specialization 'std::make_shared<phylanx::execution_tree::primitives::generic_function<locality_id_action>,
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, const std::__cxx11::basic_string<char> &, const std::__cxx11::basic_string<char> &>' requested here
>             std::make_shared<Primitive>(std::move(args), name, codename));
>                  ^
> /phylanx/tests/unit/distributed/remote_run.cpp:39:14: note: in instantiation of function template specialization 'phylanx::execution_tree::create_primitive<phylanx::execution_tree::primitives::generic_function<locality_id_action> >' requested here
>             &create_primitive<generic_function<locality_id_action>>,
>              ^
> /usr/local/include/hpx/util/tuple.hpp:447:48: note: candidate template ignored: requirement 'std::conditional<detail::pack<vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char>, eval_context>::size == 0, typename
>       std::enable_if<!std::is_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, typename
>       std::decay<vector<primitive_argument_type, allocator<primitive_argument_type> > >::type>::value && !detail::are_tuples_compatible_not_same<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type,
>       allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, vector<primitive_argument_type, allocator<primitive_argument_type> > &&>::value, detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type>
>       >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type, allocator<primitive_argument_type> > > &&> >::type,
>       detail::are_tuples_compatible<tuple<vector<primitive_argument_type, allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char> >, tuple<vector<primitive_argument_type,
>       allocator<primitive_argument_type> >, vector<primitive_argument_type, allocator<primitive_argument_type> >, basic_string<char>, basic_string<char>, eval_context> &&> >::type::value' was not satisfied [with U =
>       std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type> >, Us = <std::vector<phylanx::execution_tree::primitive_argument_type, std::allocator<phylanx::execution_tree::primitive_argument_type>
>       >, std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, phylanx::execution_tree::eval_context>]
>         explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(U&& v, Us&&... vs)
>                                                ^
> /usr/local/include/hpx/util/tuple.hpp:416:39: note: candidate constructor template not viable: requires 0 arguments, but 5 were provided
>         HPX_CONSTEXPR HPX_HOST_DEVICE tuple()
>                                       ^
> /usr/local/include/hpx/util/tuple.hpp:488:39: note: candidate constructor template not viable: requires single argument 'other', but 5 arguments were provided
>         HPX_CONSTEXPR HPX_HOST_DEVICE tuple(UTuple&& other)
>                                       ^
> /usr/local/include/hpx/util/tuple.hpp:423:48: note: candidate constructor not viable: requires 4 arguments, but 5 were provided
>         explicit HPX_CONSTEXPR HPX_HOST_DEVICE tuple(Ts const&... vs)
>                                                ^
> /usr/local/include/hpx/util/tuple.hpp:455:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
>         tuple(tuple const&) = default;
>         ^
> /usr/local/include/hpx/util/tuple.hpp:460:9: note: candidate constructor not viable: requires 1 argument, but 5 were provided
>         tuple(tuple&&) = default;
>         ^
> 4 errors generated.
> tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/build.make:62: recipe for target 'tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/remote_run.cpp.o' failed
> make[3]: *** [tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/remote_run.cpp.o] Error 1
> CMakeFiles/Makefile2:9632: recipe for target 'tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/all' failed
> make[2]: *** [tests/unit/distributed/CMakeFiles/remote_run_test_exe.dir/all] Error 2
> CMakeFiles/Makefile2:300: recipe for target 'CMakeFiles/tests.dir/rule' failed
> make[1]: *** [CMakeFiles/tests.dir/rule] Error 2
> Makefile:266: recipe for target 'tests' failed
> make: *** [tests] Error 2
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/STEllAR-GROUP/phylanx/issues/674>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAgQf87hOwaKJ0bBbVcW-b30f7FD96T8ks5uzv1CgaJpZM4Y4dFS>.
> 

--
Kevin Huck, PhD
Research Associate / Computer Scientist
OACISS - Oregon Advanced Computing Institute for Science and Society
University of Oregon
khuck@cs.uoregon.edu <mailto:khuck@cs.uoregon.edu>
http://tau.uoregon.edu <http://tau.uoregon.edu/>

--- a/tests/unit/distributed/remote_run.cpp
+++ b/tests/unit/distributed/remote_run.cpp
@@ -23,7 +23,7 @@ namespace phylanx { namespace execution_tree { namespace primitives
     hpx::future<primitive_argument_type> locality_id(
         phylanx::execution_tree::primitive_arguments_type const&,
         phylanx::execution_tree::primitive_arguments_type const&,
-        std::string const&, std::string const&);
+        std::string const&, std::string const&, eval_context ctx);
 }}}
 
 HPX_PLAIN_ACTION(
@@ -48,7 +48,7 @@ namespace phylanx { namespace execution_tree { namespace primitives
     hpx::future<primitive_argument_type> locality_id(
         phylanx::execution_tree::primitive_arguments_type const&,
         phylanx::execution_tree::primitive_arguments_type const&,
-        std::string const&, std::string const&)
+        std::string const&, std::string const&, eval_context)
     {
         std::int64_t locality_ =
             hpx::naming::get_locality_id_from_id(hpx::find_here());

